packages/medusa/src/strategies/batch-jobs/product/export.ts

```diff
@@ -0,0 +1,344 @@
+            lines.forEach((line) => writeStream.write(line))
+          })
+
+          advancementCount += products.length
+          offset += products.length
+          products.length = 0
+
+          batchJob = (await this.batchJobService_
+            .withTransaction(transactionManager)
+            .update(batchJobId, {
+              result: {
+                file_key: fileKey,
+                count: productCount,
+                advancement_count: advancementCount,
+                progress: advancementCount / productCount,
+              },
+            })) as ProductExportBatchJob
+
+          if (batchJob.status === BatchJobStatus.CANCELED) {
+            writeStream.end()
+
+            await this.fileService_
+              .withTransaction(transactionManager)
+              .delete({ key: fileKey })
+            return
+          }
+        }
+
+        writeStream.end()
+
+        return await promise
+      },
+      "REPEATABLE READ",
+      async (err) =>
+        this.handleProcessingError(batchJobId, err, {
+          count: productCount,
+          advancement_count: advancementCount,
+          progress: advancementCount / productCount,
+        })
+    )
+  }
+
+  public async buildHeader(batchJob: ProductExportBatchJob): Promise<string> {
+    const {
+      dynamicImageColumnCount,
+      dynamicMoneyAmountColumnCount,
+      dynamicOptionColumnCount,
+    } = batchJob.context.shape
+
+    this.appendOptionsDescriptors(dynamicOptionColumnCount)
+    this.appendMoneyAmountDescriptors(dynamicMoneyAmountColumnCount)
+    this.appendImagesDescriptors(dynamicImageColumnCount)
+
+    return [...this.columnDescriptors.keys(), this.NEWLINE_].join(
+      this.DELIMITER_
+    )
+  }
+
+  private appendImagesDescriptors(maxImagesCount: number): void {
+    for (let i = 0; i < maxImagesCount; ++i) {
+      this.columnDescriptors.set(`Image ${i + 1} Url`, {
+        accessor: (product: Product) => product?.images[i]?.url ?? "",
+        entityName: "product",
+      })
+    }
+  }
+
+  private appendOptionsDescriptors(maxOptionsCount: number): void {
+    for (let i = 0; i < maxOptionsCount; ++i) {
+      this.columnDescriptors
+        .set(`Option ${i + 1} Name`, {
+          accessor: (productOption: Product) =>
+            productOption?.options[i]?.title ?? "",
+          entityName: "product",
+        })
+        .set(`Option ${i + 1} Value`, {
+          accessor: (variant: ProductVariant) =>
+            variant?.options[i]?.value ?? "",
+          entityName: "variant",
+        })
+    }
+  }
+
+  private appendMoneyAmountDescriptors(maxMoneyAmountCount: number): void {
+    for (let i = 0; i < maxMoneyAmountCount; ++i) {
+      this.columnDescriptors
+        .set(`Price ${i + 1} Currency code`, {
+          accessor: (variant: ProductVariant) =>
+            variant?.prices[i]?.currency_code?.toLowerCase() ?? "",
+          entityName: "variant",
+        })
+        .set(`Price ${i + 1} Region name`, {
```

Comment:
fix: this should be Price Australia, Price France, etc. `"Region name"` is just a placeholder.